From da887d179bff6597355b0214c4e6c56f327b6ce3 Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Mon, 2 Aug 2021 13:47:36 +0200
Subject: [PATCH] Make sure to actually commit GSettings changes

g_settings_set_* is async, so make sure the change is actually performed before
returning and eventually exiting. This mostly affected the gtk_theme update
binary, because that isn't long running unlike the kded module.
---
 kded/configeditor.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/kded/configeditor.cpp b/kded/configeditor.cpp
index a51f48b..587ff7c 100644
--- a/kded/configeditor.cpp
+++ b/kded/configeditor.cpp
@@ -38,12 +38,15 @@ void ConfigEditor::setGtkConfigValueGSettings(const QString &paramName, const QV
     } else if (paramValue.type() == QVariant::Type::Bool) {
         g_settings_set_boolean(gsettings, paramName.toUtf8().constData(), paramValue.toBool());
     }
+
+    g_settings_sync();
 }
 
 void ConfigEditor::setGtkConfigValueGSettingsAsEnum(const QString &paramName, int paramValue, const QString &category)
 {
     g_autoptr(GSettings) gsettings = g_settings_new(category.toUtf8().constData());
     g_settings_set_enum(gsettings, paramName.toUtf8().constData(), paramValue);
+    g_settings_sync();
 }
 
 void ConfigEditor::setGtkConfigValueSettingsIni(const QString &versionString, const QString &paramName, const QVariant &paramValue)
-- 
2.32.0

